spec/index.bs

```diff
@@ -2517,6 +2518,82 @@ dictionary GPUImageBitmapCopyView {
+
+<div algorithm="GPUCommandEncoder.copyBufferToTexture">
+
+  **Arguments:**
+    - {{GPUBufferCopyView}} |source|
+    - {{GPUTextureCopyView}} |destination|
+    - {{GPUExtent3D}} |copySize|
+
+  **Returns:** void
+
+  Encode a command into the {{GPUCommandEncoder}} that copies |copySize| texels from a sub-region of a {{GPUBuffer}} |source| to another sub-region of a {{GPUTexture}} [=subresource=] |destination|.
+
+</div>
+
+<div algorithm class=validusage>
+
+<dfn abstract-op>copyBufferToTexture Valid Usage</dfn>
+
+Given a {{GPUCommandEncoder}} |encoder| and the arguments {{GPUBufferCopyView}} |source|, {{GPUTextureCopyView}} |destination|, {{GPUExtent3D}} |copySize|, the following validation rules apply:
+
+  - |encoder| must be a [=valid=] {{GPUCommandEncoder}}.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPURenderPassEncoder}} is active on |encoder|.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPUComputePassEncoder}} is active on |encoder|.
+  - |source|.{{GPUBufferCopyView/buffer}} must be a [=valid=] {{GPUBuffer}}.
+  - |destination|.{{GPUTextureCopyView/texture}} must be a [=valid=] {{GPUTexture}}.
+  - The {{GPUBuffer/[[usage]]}} of |source|.{{GPUBufferCopyView/buffer}} must contain {{GPUBufferUsage/COPY_SRC}}.
+  - The {{GPUTexture/[[textureUsage]]}} of |destination|.{{GPUTextureCopyView/texture}} must contain {{GPUTextureUsage/COPY_DST}}.
+  - The {{GPUTexture/[[sampleCount]]}} of |destination|.{{GPUTextureCopyView/texture}} must be 1.
+  - If |source|.{{GPUBufferCopyView/rowsPerImage}} is not 0, it must be greater than or equal to |copySize|.{{GPUExtent3DDict/height}}.
+  - |source|.{{GPUBufferCopyView/rowsPerImage}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} must be a multiple of [=texel block width=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/width}} must be a multiple of [=texel block width=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/height}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |source|.{{GPUBufferCopyView/bytesPerRow}} must be a multiple of 256.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is not 0, it must be greater than or equal to the number of |bytes in a complete row|.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is 0, the number of |bytes in a complete row| must be a multiple of 256.
```

Comment:
I think we were not going to allow bytesPerRow=0 with width>1, or we would have made it optional (although I don't remember when we decided that). In any case, that's what's implied by the text before this PR, so let's keep that semantic for now, and IMO match the semantic of rowsPerImage:

```suggestion
  - If |copySize|.[=Extent3D/height=] > 1:
      - |source|.{{GPUBufferCopyView/bytesPerRow}} must be > 0.
```

Reply:
May be able to simplify if trueBytesPerRow is predefined